src/components/Heroes/Heroes.jsx

```diff
@@ -3,29 +3,46 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import heroes from 'dotaconstants/json/heroes.json';
 import { Tabs, Tab } from 'material-ui/Tabs';
+import TextField from 'material-ui/TextField';
 import HeroList from './HeroList';
 import style from './Heroes.css';
 import Ranking from './Ranking';
 import Benchmark from './Benchmark';
 import HeroBadge from './HeroBadge';
 
-const createHeroList = (heroes) => {
-  const result = Object.keys(heroes)
-    .map(key => ({ ...heroes[key], img: API_HOST + heroes[key].img }))
-    .sort((a, b) => (a.localized_name.localeCompare(b.localized_name)));
-  return result;
+const createHeroList = (heroes, filter) => {
+  const filteredHeroes = [];
+
+  Object.keys(heroes).forEach((key) => {
+    if (typeof filter === 'undefined' || filter.length === 0) {
```

Comment:
[not quite the same](https://jsfiddle.net/jrdsz5cn/1/)

`!filter` !== `typeof filter === 'undefined'`